Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -86,17 +93,55 @@ public override string DataSource
             }
         }
 
-        public override string ServerVersion => _connectionState == ConnectionState.Open ? SfSession.serverVersion : "";
+        public override string ServerVersion => IsOpen() ? SfSession.serverVersion : String.Empty;
 
         public override ConnectionState State => _connectionState;
 
+        private string GetTransactionId()

```

Comment:
will this cause delay in closing the connection? Is there a way we can keep transaction status in the driver without executing a SQL command? Will this execute only when connection pool is enable?

Reply:
I'll change it to execute only when pool is enabled.
It does introduce a delay since there is a call to the database. In theory we could avoid it and store the transaction reference in the connection. However, if one decides to start a transaction with SnowflakeDbConnection.ExecuteNonQuery the connection will never be aware of such a transaction nor have a reference to the transaction object. 
We could of course track such commands within ExecuteNonQuery and some regexp and force users to use SnowflakeDbConnection.BeginDbTransaction but it would affect current apps as the breaking change.

Reply:
Keeping transaction status is a tricky thing. We can think about it in the future.

Reply:
Can we at least check for auto commit on/off and if auto commit is ON then no need to check for txn id or rolling back the transaction. Most of the driver has API for auto commit ON or OFF. e.g. in JDBC it's setAutoCommit(boolean) API.

Reply:
When autocommit is enabled and user starts an explicit transaction we might still need to check if this has been rolled back or committed:
"autocommit determines whether a DML statement, when executed without an active transaction, is automatically committed after the statement successfully completes. ..."
Autocommit can be also driven by account settings in which case we need to query database to get this setting as the state might not always be achieved with connection string/explicit statement?

Reply:
As a result of our discussions only ROLLBACK will be submitted. I removed this method.